docs/okeps/okep-5094-layer2-transit-router.md

```diff
@@ -0,0 +1,870 @@
+        end
+    end
+    
+    subgraph node3["node3 <b>(old topo)</b>"]
+        subgraph VM["Virtual Machine"]
+            class VM vmStyle;
+            term["default gw 
+            203.203.0.1
+            (0a:58:64:41:00:04)"]
+        end
+        subgraph GR-node3
+            rtos-GR-node3["rtos-GR-node3  203.203.0.1/24 100.65.0.4/16 (0a:58:64:41:00:04)"]
+        end
+        subgraph layer2-switch-node3["layer2-switch"]
+            stor-GR-node1["stor-GR-node1
+                100.65.0.2/16 (0a:58:64:41:00:02) 
+                type: remote"]
+            stor-GR-node3-local["stor-GR-node3 
+                type: router"]
+        end
+    end
+    
+    rtotr-GR-node1 <--> trtor-GR-node1
+    rtotr-GR-node2 <--> trtor-GR-node2
+    rtos-GR-node3 <--> stor-GR-node3-local
+    stor-GR-node3-local <--> stor-GR-node3
+    rtos-layer2-switch <--> stor-ovn_cluster_router
+    stor-GR-node1 <--> stor-ovn_cluster_router
+    VM <--> layer2-switch-node3
+    
+    class VM vmStyle;
+    class rtotr-GR-node1 portStyle;
+    class rtotr-GR-node2 portStyle;
+    class rtotr-GR-node3 portStyle;
+    class trtor-GR-node1 portStyle;
+    class trtor-GR-node2 portStyle;
+    class trtor-GR-node3 portStyle;
+    class stor-GR-node3 portStyle;
+    class rtos-GR-node3 portStyle;
+    class stor-ovn_cluster_router portStyle;
+    class rtos-layer2-switch portStyle;
+    class stor-GR-node3-local portStyle;
+    class stor-GR-node1 portStyle;
+    class GR-node1 routerStyle;
+    class GR-node2 routerStyle;
+    class GR-node3 routerStyle;
+    class ovn_cluster_router routerStyle;
+    class layer2-switch switchStyle;
+    class layer2-switch-node3 switchStyle;
+    class term termStyle;
+    class node1,node2,node3 nodeStyle;
+```
+
+With this topology updated nodes should keep the remote node connection towards
+not updated nodes using the original switch remote LSPs, since the old nodes
+has not transit router yet and after detecting that the node has the new topology
+"move" that LSP to the transit router as a LRP pointing to the chassis and
+tunnel key.
+
+Anodher change is that the transit router LRP that connects to the switch has 
+to keep the join IP since remote switch LSPs will still trigger L2 traffic
+towards it, after all the remote nodes has upgrade to transit router this
+join IP should be removed since it's already at the gateway router.
+
+Also the mechanism on how to check if remote node is still not updated can be
+implemented by checking the transit router subnet annotation, since it should
+pop up only if the node is prepare for it.
+
+
+### Testing Details
+
+* Unit test checking topology will need to be adapted.
+* E2e tests should be the same and pass since this is a refactoring not adding or removing features.
+* Scale tests to check if adding the transit router affects it
+* Upgrade tests are the most important in this case. The tests should perform an upgrade
```

Comment:
can we have an e2e upstream to test upgrade of a layer 2 udn and measure traffic outage using iperf?

Reply:
I think there are some upgrade e2e already upstrea, I haven't play with it though.